core/src/main/java/io/parsingdata/metal/expression/value/reference/Self.java

```diff
@@ -35,8 +31,8 @@
 public class Self implements ValueExpression {
 
     @Override
-    public ImmutableList<Optional<Value>> eval(final ParseState parseState, final Encoding encoding) {
-        return ImmutableList.create(parseState.order.current().map(identity()));
+    public ImmutableList<Value> eval(final ParseState parseState, final Encoding encoding) {
+        return parseState.order.current().map(ImmutableList::<Value>create).orElseGet(ImmutableList::new);
```

Comment:
I like the way you moved it to three lines in CurrentIteration. I would prefer to keep that consistent and also move it to three lines here.

Reply:
Both `Self` and `CurrentIteration` have been changed in the other two PRs quite a bit, but thanks for pointing me towards `CurrentIteration`! It turns out I forgot to convert its `level` operand to a `SingleValueExpression`!